k8s/pkg/proxyregistryserver/discovery.go

```diff
@@ -70,9 +70,14 @@ func (d *discoveryService) FindNetworkService(ctx context.Context, request *regi
 		if dErr != nil {
 			return nil, dErr
 		}
-		for _, nsm := range response.NetworkServiceManagers {
+		managers := make(map[string]*registry.NetworkServiceManager)
+		for key, nsm := range response.NetworkServiceManagers {
+			managers[key] = nsm
 			if url, urlErr := d.currentDomainNSMgrURL(ctx, d.clusterInfoService, nsm.Url); urlErr == nil && nsm.Url == url {
-				return d.handleLocalFindCase(response, url), nil
+				d.handleLocalFindCase(response, nsm, url)
+				delete(managers, key)
```

Comment:
I think you could just move the add (line 75) to after the if statement and remove this delete, right?

Reply:
Thanks, fixed.